### ISSUE TITLE
No connection after upgrade to Valetudo 0.5.2

### ISSUE THREAD

**Issue by Muegic** [iobroker-community-adapters/ioBroker.mihome-vacuum#146]
Hi,
I upgraded my Gen1 to Valetudo 0.5.2 due to random firmware resets and now I can´t control the cleaner anymore.
I tried both Tokens from the Valetudo Settings page (Cloud token & Local token) but none seems to work.
With the old Valetudo version (I think it was 0.3.1) I had only one Token, which worked fine with your adapter.

Adapter is green and I get the all the data points like start, pause,...  but they don´t control the cleaner.
I also upgraded the Adapter to version 2.0.7 but still the same problem.

This is the log:

mihome-vacuum.1 | 2020-06-04 00:07:08.094 | info | (20385) Disconnect
-- | -- | -- | --
mihome-vacuum.1 | 2020-06-04 00:06:46.932 | info | (20385) Connected
mihome-vacuum.1 | 2020-06-04 00:06:46.919 | info | (20385) Disconnect
mihome-vacuum.1 | 2020-06-04 00:06:18.065 | info | (20385) Connected
mihome-vacuum.1 | 2020-06-04 00:06:18.053 | info | (20385) Disconnect
mihome-vacuum.1 | 2020-06-04 00:05:56.911 | info | (20385) Connected
mihome-vacuum.1 | 2020-06-04 00:05:56.895 | info | (20385) Disconnect
mihome-vacuum.1 | 2020-06-04 00:05:28.050 | info | (20385) Connected
mihome-vacuum.1 | 2020-06-04 00:05:28.029 | info | (20385) Disconnect
mihome-vacuum.1 | 2020-06-04 00:05:06.883 | info | (20385) Connected
mihome-vacuum.1 | 2020-06-04 00:05:06.868 | info | (20385) Disconnect
mihome-vacuum.1 | 2020-06-04 00:04:56.015 | info | (20385) should trigger cleaning all null, but is currently active. Add to queue
mihome-vacuum.1 | 2020-06-04 00:04:47.209 | info | (20385) should trigger cleaning all null, but is currently active. Add to queue
mihome-vacuum.1 | 2020-06-04 00:04:46.077 | info | (20385) Connected
mihome-vacuum.1 | 2020-06-04 00:04:46.066 | info | (20385) Disconnect
mihome-vacuum.1 | 2020-06-04 00:04:44.063 | info | (20385) trigger cleaning all null
mihome-vacuum.1 | 2020-06-04 00:04:35.515 | warn | (20385) no answer received after after 3 times -> pause miIO.info from request parameters, try again in one hour
mihome-vacuum.1 | 2020-06-04 00:04:21.398 | info | (20385) set nächster Timer: Nicht verfügbar
mihome-vacuum.1 | 2020-06-04 00:04:20.486 | warn | (20385) Time difference between Mihome Vacuum and ioBroker: 2 sec
mihome-vacuum.1 | 2020-06-04 00:04:20.484 | info | (20385) Connected
mihome-vacuum.1 | 2020-06-04 00:04:20.438 | info | (20385) Disconnect
mihome-vacuum.1 | 2020-06-04 00:04:20.420 | info | (20385) Create state clean_home for controlling by cloud adapter
mihome-vacuum.1 | 2020-06-04 00:04:20.417 | info | (20385) Expert mode disabled, states deleted
mihome-vacuum.1 | 2020-06-04 00:04:20.317 | info | (20385) starting. Version 2.0.7 in /opt/iobroker/node_modules/iobroker.mihome-vacuum, node: v12.17.0
mihome-vacuum.1 | 2020-06-04 00:03:52.914 | info | (20345) Terminated (START_IMMEDIATELY_AFTER_STOP): Without reason
mihome-vacuum.1 | 2020-06-04 00:03:52.912 | info | (20345) terminating
mihome-vacuum.1 | 2020-06-04 00:03:52.908 | info | (20345) Got terminate signal TERMINATE_YOURSELF
mihome-vacuum.1 | 2020-06-04 00:03:14.632 | info | (20345) Connected
mihome-vacuum.1 | 2020-06-04 00:03:14.623 | info | (20345) Disconnect

Any ideas?
Is Valetudo 0.5.2 working with your adapter, or do I need to downgrade Valetudo again?

Thanks in advance

**Comment by Muegic** [comment]
Hi Dirk, here is the debug log:

mihome-vacuum.1 | 2020-06-04 20:12:25.210 | info | (23370) Disconnect
-- | -- | -- | --
mihome-vacuum.1 | 2020-06-04 20:12:20.614 | debug | (23370) no answer for get_sound_volume(id:16) received, giving up
mihome-vacuum.1 | 2020-06-04 20:12:20.411 | debug | (23370) no answer for get_clean_summary(id:15) received, giving up
mihome-vacuum.1 | 2020-06-04 20:12:20.215 | debug | (23370) no answer for get_status(id:14) received, giving up
mihome-vacuum.1 | 2020-06-04 20:12:20.208 | debug | (23370) no answer for get_consumable(id:13) received, giving up
mihome-vacuum.1 | 2020-06-04 20:12:05.192 | debug | (23370) requesting params every: 60 Sec
mihome-vacuum.1 | 2020-06-04 20:12:00.635 | debug | (23370) no answer for get_sound_volume(id:12) received, giving up
mihome-vacuum.1 | 2020-06-04 20:12:00.432 | debug | (23370) no answer for get_clean_summary(id:11) received, giving up
mihome-vacuum.1 | 2020-06-04 20:12:00.233 | debug | (23370) no answer for get_consumable(id:10) received, giving up
mihome-vacuum.1 | 2020-06-04 20:12:00.225 | debug | (23370) no answer for get_status(id:9) received, giving up
mihome-vacuum.1 | 2020-06-04 20:11:45.215 | debug | (23370) requesting params every: 60 Sec
mihome-vacuum.1 | 2020-06-04 20:11:45.213 | info | (23370) Connected
mihome-vacuum.1 | 2020-06-04 20:11:45.199 | info | (23370) Disconnect
mihome-vacuum.1 | 2020-06-04 20:11:40.210 | debug | (23370) no answer for get_status(id:8) received, giving up
mihome-vacuum.1 | 2020-06-04 20:11:23.213 | debug | (23370) no answer for get_room_mapping(id:7) received, giving up
mihome-vacuum.1 | 2020-06-04 20:11:23.208 | debug | (23370) no answer for get_carpet_mode(id:6) received, giving up
mihome-vacuum.1 | 2020-06-04 20:11:20.984 | debug | (23370) no answer for get_sound_volume(id:5) received, giving up
mihome-vacuum.1 | 2020-06-04 20:11:20.977 | debug | (23370) no answer for get_clean_summary(id:4) received, giving up
mihome-vacuum.1 | 2020-06-04 20:11:20.455 | debug | (23370) no answer for get_consumable(id:3) received, giving up
mihome-vacuum.1 | 2020-06-04 20:11:20.256 | warn | (23370) no answer received after after 3 times -> pause miIO.info from request parameters, try again in one hour
mihome-vacuum.1 | 2020-06-04 20:11:20.253 | debug | (23370) no answer for miIO.info(id:2) received, giving up
mihome-vacuum.1 | 2020-06-04 20:11:20.247 | debug | (23370) no answer for get_status(id:1) received, giving up
mihome-vacuum.1 | 2020-06-04 20:11:06.180 | info | (23370) set nächster Timer: Nicht verfügbar
mihome-vacuum.1 | 2020-06-04 20:11:05.631 | debug | (23370) 25719
mihome-vacuum.1 | 2020-06-04 20:11:05.236 | debug | (23370) requesting params every: 60 Sec
mihome-vacuum.1 | 2020-06-04 20:11:05.229 | info | (23370) Connected
mihome-vacuum.1 | 2020-06-04 20:11:05.196 | debug | (23370) server started on 0.0.0.0:53422
mihome-vacuum.1 | 2020-06-04 20:11:05.190 | info | (23370) Disconnect
mihome-vacuum.1 | 2020-06-04 20:11:05.172 | info | (23370) Create state clean_home for controlling by cloud adapter
mihome-vacuum.1 | 2020-06-04 20:11:05.170 | info | (23370) Expert mode disabled, states deleted
mihome-vacuum.1 | 2020-06-04 20:11:05.156 | debug | (23370) load Map creator... true
mihome-vacuum.1 | 2020-06-04 20:11:05.061 | info | (23370) starting. Version 2.0.7 in /opt/iobroker/node_modules/iobroker.mihome-vacuum, node: v12.17.0
mihome-vacuum.1 | 2020-06-04 20:11:02.375 | debug | (23370) statesDB connected
mihome-vacuum.1 | 2020-06-04 20:11:02.373 | debug | (23370) States connected to redis: 127.0.0.1:9000
mihome-vacuum.1 | 2020-06-04 20:11:02.343 | debug | (23370) States create System PubSub Client
mihome-vacuum.1 | 2020-06-04 20:11:02.338 | debug | (23370) States create User PubSub Client
mihome-vacuum.1 | 2020-06-04 20:11:02.320 | debug | (23370) Redis States: Use Redis connection: 127.0.0.1:9000
mihome-vacuum.1 | 2020-06-04 20:11:02.317 | debug | (23370) objectDB connected
mihome-vacuum.1 | 2020-06-04 20:11:02.308 | debug | (23370) Objects connected to redis: 127.0.0.1:9001
mihome-vacuum.1 | 2020-06-04 20:11:02.278 | debug | (23370) Objects client initialize lua scripts
mihome-vacuum.1 | 2020-06-04 20:11:02.277 | debug | (23370) Objects create User PubSub Client
mihome-vacuum.1 | 2020-06-04 20:11:02.275 | debug | (23370) Objects create System PubSub Client
mihome-vacuum.1 | 2020-06-04 20:11:02.270 | debug | (23370) Objects client ready ... initialize now
mihome-vacuum.1 | 2020-06-04 20:11:01.840 | debug | (23370) Redis Objects: Use Redis connection: 127.0.0.1:9001
mihome-vacuum.1 | 2020-06-04 20:10:58.420 | info | (20385) Terminated (START_IMMEDIATELY_AFTER_STOP): Without reason
mihome-vacuum.1 | 2020-06-04 20:10:58.419 | info | (20385) terminating
mihome-vacuum.1 | 2020-06-04 20:10:58.417 | info | (20385) Got terminate signal TERMINATE_YOURSELF
mihome-vacuum.1 | 2020-06-04 20:10:53.985 | info | (20385) Connected
mihome-vacuum.1 | 2020-06-04 20:10:53.982 | info | (20385) Disconnect

I don´t think the reconnects every 40 sec are a WLAN issue - at least I don´t notice WLAN issues elsewhere.

**Comment by dirkhe** [comment]
Try to go Into mihome App for some minutes. Perhaps Restart adapter. 
If this not work, try to get the token via the Adapter

**Comment by Muegic** [comment]
Unfortunately I removed the robot from the mihome app about a year ago.
And I was not able to bring it back into the mihome app with a rooted firmware (I tried iOS & two older Android apks and several Valetudo versions).

Today I flashed the cleaner with Valetudo 0.5.1 & 0.5.0 and tried both tokens (Cloud & local) from Valetudo - nothing worked.
I finally flashed Valetudo 0.4.0, took the one token from Valetudo and the adapter worked flawless.

Then I reset the cleaner to the original firmware and connected it with mihome in iOS successfully.
I reinstalled the adapter and was able to get the new token through the adapter from the cloud.

I flashed Valetudo 0.5.2 again and tried to get the token from the Adapter/cloud, but it was still the old token and it did not work.
When I try to access the cleaner through the mihome app, it gives me two options: update the firmware or Quit.

I´m quite fed up now, and go back to Valetudo 0.4.0... 

Trotzdem Danke dass du es versucht hast ;-)
